kmir/src/kmir/kmir.py

```diff
@@ -11,17 +11,60 @@
 from typing import Final, Optional, Union, final
 
 from pyk.cli.utils import check_dir_path, check_file_path
-from pyk.kast.inner import KInner
+from pyk.cterm import CTerm
+from pyk.kast.inner import KApply, KInner, KSequence, KVariable
+from pyk.kcfg.semantics import KCFGSemantics
 from pyk.ktool.kprint import KAstInput, KAstOutput, _kast, gen_glr_parser
 from pyk.ktool.kprove import KProve
 from pyk.ktool.krun import KRunOutput, _krun
+from pyk.prelude.k import K
 from pyk.utils import BugReport
 
 from .preprocessor import preprocess
 
 _LOGGER: Final = logging.getLogger(__name__)
 
 
+class KMIRSemantics(KCFGSemantics):
+    def is_terminal(self, cterm: CTerm) -> bool:
+        k_cell = cterm.cell('K_CELL')
+        # <k> #halt </k>
+        if k_cell == KMIR.halt():
+            return True
+        elif type(k_cell) is KSequence:
+            # <k> . </k>
+            if k_cell.arity == 0:
+                return True
+            # <k> #halt </k>
+            elif k_cell.arity == 1 and k_cell[0] == KMIR.halt():
+                return True
+            elif (
+                k_cell.arity == 2 and k_cell[0] == KMIR.halt() and type(k_cell[1]) is KVariable and k_cell[1].sort == K
+            ):
+                return True
+        return False
+
+    @staticmethod
+    def terminal_rules() -> list[str]:
+        terminal_rules = ['EVM.halt']
```

Comment:
This looks suspicious to me; I presume it should be `MIR.halt` looking at the rest of the code?